nvcodec/README.md

```diff
@@ -0,0 +1,136 @@
+
+ * https://developer.nvidia.com/nvidia-video-codec-sdk/download
+
+Or, if you want a simple test, use the sample code below.
+
+We can use [Maven 3](http://maven.apache.org/) to download and install automatically all the class files as well as the native binaries. To run this sample code, after creating the `pom.xml` and `SampleEncodeDecode.java` source files below, simply execute on the command line:
+```bash
+ $ mvn compile exec:java
+```
+
+### The `pom.xml` build file
+```xml
+<project>
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.bytedeco.nvcodec</groupId>
+    <artifactId>sampleencodedecode</artifactId>
+    <version>1.5.6</version>
+    <properties>
+        <exec.mainClass>SampleEncodeDecode</exec.mainClass>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.bytedeco</groupId>
+            <artifactId>nvcodec-platform</artifactId>
+            <version>11.0.10-1.5.6</version>
+        </dependency>
+
+        <!-- Additional dependencies to use bundled CUDA -->
+        <dependency>
+            <groupId>org.bytedeco</groupId>
+            <artifactId>cuda-platform-redist</artifactId>
+            <version>11.2-8.1-1.5.6</version>
+        </dependency>
+    </dependencies>
+    <build>
+        <sourceDirectory>.</sourceDirectory>
+    </build>
+</project>
+```
+
+### The `SampleEncodeDecode.java` source file
+```java
+import org.bytedeco.cuda.cudart.CUctx_st;
+import org.bytedeco.javacpp.IntPointer;
+import org.bytedeco.nvcodec.nvcuvid.CUVIDDECODECAPS;
+
+import static org.bytedeco.cuda.global.cudart.*;
+import static org.bytedeco.nvcodec.global.nvcuvid.*;
+import static org.bytedeco.nvcodec.global.nvencodeapi.*;
+
+public class SampleEncodeDecode {
+    public static _NVENCSTATUS getNVENCSTATUSByValue(int value) {
+        _NVENCSTATUS[] statuses = _NVENCSTATUS.values();
+
+        for (_NVENCSTATUS status : statuses) {
+            if (status.value == value) {
+                return status;
+            }
+        }
+
+        return null;
+    }
+    public static void checkCudaApiCall(String functionName, int result) {
+        if (result != 0) {
+            System.err.printf("ERROR: %s returned '%d' \r\n", functionName, result);
+            System.exit(-1);
+        }
+    }
+
+    public static void main(String[] args) {
+        int targetGpu = 0; // Change to your own NVIDIA GPU number.
+
+        CUctx_st cuContext = new CUctx_st();
+
+        checkCudaApiCall("cuInit", cuInit(0));
+        checkCudaApiCall("cuCtxCreate", cuCtxCreate(cuContext, 0, targetGpu));
+
+        // Encoder max supported version check.
+        {
+            IntPointer version = new IntPointer(1);
+            _NVENCSTATUS status = getNVENCSTATUSByValue(NvEncodeAPIGetMaxSupportedVersion(version));
```

Comment:
That's what the `intern()` method is for. Do this instead:
```java
 _NVENCSTATUS status = NvEncodeAPIGetMaxSupportedVersion(version).intern();
```
BTW, since this is a C API, not C++, it may be better not to use `Info.enumerate()`.
That way you wouldn't need to call `value` and/or `intern()` everywhere like that.

Reply:
` _NVENCSTATUS status = NvEncodeAPIGetMaxSupportedVersion(version).intern();` does not work.

so i changed like that
```
        {
            IntPointer version = new IntPointer(1);
            int status = NvEncodeAPIGetMaxSupportedVersion(version);

            if (status != _NVENCSTATUS.NV_ENC_SUCCESS.value) {
                System.err.printf("ERROR: NvEncodeAPIGetMaxSupportedVersion returned '%s' \r\n", status);
                System.exit(-1);
            } else {
                System.out.printf("Encoder Max Supported Version\t : %d \r\n", version.get());
            }
        }
```

is this method okay?

Reply:
Ah.. you mean change the parse code. ok i will try

Reply:
Ah, I see, we'd need to make sure that `NVENCSTATUS` maps to `_NVENCSTATUS`. In any case, it's probably a better idea to get rid of `Info.enumerate()`. Try to remove it and let me know what you think.

Reply:
```
typedef enum _NVENCSTATUS
{
    NV_ENC_SUCCESS,
...
    NV_ENC_ERR_OUT_OF_MEMORY,

    NV_ENC_ERR_RESOURCE_NOT_MAPPED,
} NVENCSTATUS;
```

Here is original NVENCSTATUS c code

Reply:
Ok i will remove enumerate info code and than check

Reply:
You were right! It has been changed to more correct. i pushed!